src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenReadOnlySpanConstructionTest.cs

```diff
@@ -2685,5 +2652,779 @@ public static int M()
+  IL_001d:  newobj     "System.ReadOnlySpan<string>..ctor(string[])"
+  IL_0022:  ret
+}
+""");
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69472")]
+        public void ReadOnlySpanFromArrayOfConstants_OtherStrings_MissingReadOnlySpanCtor()
+        {
+            var src = """
+var values = C.M();
+System.Console.Write($"{values.Length} {values[0]} {values[1]}");
+
+public class C
+{
+    public static System.ReadOnlySpan<string> M()
+        => new string[] { "hello", "world" };
+}
+""";
+            var compilation = CreateCompilationWithMscorlibAndSpan(src);
+            compilation.MakeMemberMissing(WellKnownMember.System_ReadOnlySpan_T__ctor_Array);
+            var verifier = CompileAndVerify(compilation, expectedOutput: "2 hello world", verify: Verification.Skipped);
+            verifier.VerifyIL("C.M", """
+{
+  // Code size       28 (0x1c)
+  .maxstack  4
+  IL_0000:  ldc.i4.2
+  IL_0001:  newarr     "string"
+  IL_0006:  dup
+  IL_0007:  ldc.i4.0
+  IL_0008:  ldstr      "hello"
+  IL_000d:  stelem.ref
+  IL_000e:  dup
+  IL_000f:  ldc.i4.1
+  IL_0010:  ldstr      "world"
+  IL_0015:  stelem.ref
+  IL_0016:  call       "System.ReadOnlySpan<string> System.ReadOnlySpan<string>.op_Implicit(string[])"
+  IL_001b:  ret
+}
+""");
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69472")]
+        public void ReadOnlySpanFromArrayOfConstants_VariableStrings()
+        {
+            var src = """
+var x = C.M();
+System.Console.Write($"{x[0]} {x[1]}");
+
+public class C
+{
+    public static System.ReadOnlySpan<string> M()
+    {
+        var hello = "hello";
+        var world = "world";
+        return new string[] { hello, world };
+    }
+}
+""";
+            var compilation = CreateCompilationWithMscorlibAndSpan(src);
+            var verifier = CompileAndVerify(compilation, expectedOutput: "hello world", verify: Verification.Skipped);
+            verifier.VerifyIL("C.M", """
+{
+  // Code size       32 (0x20)
+  .maxstack  4
+  .locals init (string V_0, //hello
+                string V_1) //world
+  IL_0000:  ldstr      "hello"
+  IL_0005:  stloc.0
+  IL_0006:  ldstr      "world"
+  IL_000b:  stloc.1
+  IL_000c:  ldc.i4.2
+  IL_000d:  newarr     "string"
+  IL_0012:  dup
+  IL_0013:  ldc.i4.0
+  IL_0014:  ldloc.0
+  IL_0015:  stelem.ref
+  IL_0016:  dup
+  IL_0017:  ldc.i4.1
+  IL_0018:  ldloc.1
+  IL_0019:  stelem.ref
+  IL_001a:  call       "System.ReadOnlySpan<string> System.ReadOnlySpan<string>.op_Implicit(string[])"
+  IL_001f:  ret
+}
+""");
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69472")]
+        public void ReadOnlySpanFromArrayOfConstants_NoInitializer()
+        {
+            var src = """
+public class C
+{
+    public static System.ReadOnlySpan<object> M() => new object[];
+}
+""";
+            var compilation = CreateCompilationWithMscorlibAndSpan(src);
+            compilation.VerifyDiagnostics(
```

Comment:
>VerifyDiagnostics [](http://example.com/codeflow?start=24&length=17)

Is this test hitting changes in the PR, given that it fails in initial binding?

Reply:
This test doesn't reach the new code. I just wanted to be sure